chart/Graphics/Rendering/Chart/Axis/Floating.hs

```diff
@@ -108,7 +108,7 @@ scaledAxis lap rs@(minV,maxV) ps0 = makeAxis' realToFrac realToFrac
     ps        = filter isValidNumber ps0
     range []  = (0,1)
     range _   | minV == maxV = if minV==0 then (-1,1) else
-                               let d = abs (minV * 0.01) in (minV-d,maxV+d)
+                               let d = max 1e-300 (abs (minV * 0.01)) in (minV-d,maxV+d)
               | otherwise    = rs
```

Comment:
I'm not actually sure if this is the correct behavior. It seems to give me a range of `(-1e-300, 1e-300)` when `rs = (0, 0)` and the freeze came back. Could you check it for me?